### PR TITLE
vim shell evasion threat

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
https://github.com/elastic/detection-rules/issues/1864

## Summary
An existing threat identification already exists for vi binary. This issue adds the identification for vim.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/execution_vi_binary.toml [hunk 1]
```diff
@@ -25,8 +26,8 @@ type = "eql"
 
 query = '''
 sequence by host.id,process.pid with maxspan=1m
-[process where process.name == "vi" and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
-[process where process.parent.name == "vi" and process.name : ("bash", "sh")]
+[process where process.name : ("vi", "vim") and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
```
Comment (brokensound77) [inline rules/linux/execution_vi_binary.toml:29]: why the change from `:` to `==`? `==` is more accurate for this case (on Linux)

Comment (Samirbous) [reply rules/linux/execution_vi_binary.toml:29]: ++

Comment (Samirbous) [inline rules/linux/execution_vi_binary.toml:29]: for case sensitive multi-values check you can use `something in ("val1", "val2")` will match only on `val1` and `val2` and not on `vAl1` or `VAL1`

```suggestion
[process where process.name in ("vi", "vim") and process.args == "-c" and process.args in (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
```

**Review by Samirbous — APPROVED** [review]

File: rules/linux/execution_vi_binary.toml [hunk 2]
```diff
@@ -25,8 +26,8 @@ type = "eql"
 
 query = '''
 sequence by host.id,process.pid with maxspan=1m
-[process where process.name == "vi" and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
-[process where process.parent.name == "vi" and process.name : ("bash", "sh")]
+[process where process.name in ("vi", "vim") and process.args == "-c" and process.args in (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
+[process where process.parent.name : ("vi", "vim") and process.name : ("bash", "sh")]
```
Comment (DefSecSentinel) [inline rules/linux/execution_vi_binary.toml:30]:
```suggestion
[process where process.name in ("vi", "vim") and process.args == "-c" and process.args in (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
[process where process.parent.name in ("vi", "vim") and process.name in ("bash", "sh")]
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/linux/execution_vi_binary.toml [hunk 3]
```diff
@@ -25,8 +26,8 @@ type = "eql"
 
 query = '''
 sequence by host.id,process.pid with maxspan=1m
```
Comment (DefSecSentinel) [inline rules/linux/execution_vi_binary.toml:28]:
```suggestion
```

File: rules/linux/execution_vi_binary.toml [hunk 4]
```diff
@@ -25,8 +26,8 @@ type = "eql"
...
-[process where process.name == "vi" and process.args : "-c" and process.args : (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
-[process where process.parent.name == "vi" and process.name : ("bash", "sh")]
+[process where process.name in ("vi", "vim") and process.args == "-c" and process.args in (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh")]
+[process where process.parent.name in ("vi", "vim") and process.name in ("bash", "sh")]
```
Comment (DefSecSentinel) [inline rules/linux/execution_vi_binary.toml:29]:
```suggestion
process where event.type == "start" and process.parent.name in ("vi", "vim") and process.parent.args == "-c" and process.parent.args in (":!/bin/bash", ":!/bin/sh", ":!bash", ":!sh") and process.name in ("bash", "sh")
```
